libs/components/indicators/testing/src/icon/icon-harness.spec.ts

```diff
@@ -0,0 +1,205 @@
+  fixture.componentInstance.iconType = iconType;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.getIconType()).toBeResolvedTo(
+    iconType ? iconType : 'fa'
+  );
+}
+
+async function validateFixedWidth(
+  iconHarness: SkyIconHarness,
+  fixture: ComponentFixture<TestComponent>,
+  fixedWidth: boolean | undefined
+): Promise<void> {
+  fixture.componentInstance.fixedWidth = fixedWidth;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.isFixedWidth()).toBeResolvedTo(
+    fixedWidth ? fixedWidth : false
+  );
+}
+
+async function validateIconSize(
+  iconHarness: SkyIconHarness,
+  fixture: ComponentFixture<TestComponent>,
+  iconSize: string | undefined
+): Promise<void> {
+  fixture.componentInstance.size = iconSize;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.getIconSize()).toBeResolvedTo(iconSize);
+}
+
+async function validateVariant(
+  iconHarness: SkyIconHarness,
+  fixture: ComponentFixture<TestComponent>,
+  variant: string
+): Promise<void> {
+  fixture.componentInstance.variant = variant;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.getVariant()).toBeResolvedTo(variant);
+}
+
+const iconTypes = ['fa', 'skyux'];
+const variants = ['line', 'solid'];
+const sizes = ['lg', '2x', '3x', '4x', '5x'];
+
+describe('Icon harness', () => {
+  async function setupTest(options: { dataSkyId?: string } = {}): Promise<{
+    iconHarness: SkyIconHarness;
+    fixture: ComponentFixture<TestComponent>;
+    loader: HarnessLoader;
+    pageLoader: HarnessLoader;
+  }> {
+    await TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [SkyIconModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(TestComponent);
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+    const pageLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    const iconHarness: SkyIconHarness = options.dataSkyId
+      ? await loader.getHarness(
+          SkyIconHarness.with({ dataSkyId: options.dataSkyId })
+        )
+      : await loader.getHarness(SkyIconHarness);
+    fixture.detectChanges();
+
+    return { iconHarness, fixture, loader, pageLoader };
+  }
+
+  it('should return the correct icon name for all icon types and variants', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    for (const type of iconTypes) {
+      fixture.componentInstance.iconType = type;
+      for (const variant of variants) {
+        fixture.componentInstance.variant = variant;
+        await validateIconName(
+          iconHarness,
+          fixture,
+          fixture.componentInstance.iconName
+        );
+      }
+    }
+  });
+
+  it('should throw error if icon name is not set', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    fixture.componentInstance.iconName = undefined;
+    await expectAsync(iconHarness.getIconName()).toBeRejectedWithError(
+      'Icon does not exist'
```

Comment:
The error for this specific scenario should be more helpful. The consumer has added the `<sky-icon>` to their template but has forgotten a `name` (or it's set to `undefined`). We need to find a way to make this error more helpful so the consumer knows what they can do to rectify it.